File: Sorting Algorithms/Insertion Sort/InsertionSort.java
public class InsertionSort implements Sort {

    @Override
    public int[] sort(int[] array) {
        
        for (int i = 1; i < array.length; i++) { 

            int key = array[i]; 
            int j = i - 1; 

            while (j >= 0 && array[j] > key) { 
                array[j + 1] = array[j]; 
                j = j - 1; 
            } 
            array[j + 1] = key; 
        } 

        return array;
    }
    
}

File: Data Science/ShakeThatSpear/insertionSort.py
def insertionSort(arr):

    for i in range(len(arr)):
        key = arr[i]
        x = i-1

        while x >= 0 and arr[x] > key:
            arr[x + 1] = arr[x]
            x = x - 1

        arr[x + 1] = key
